app/src/app/[lng]/[inventory]/data/[step]/[subsector]/page.tsx

```diff
@@ -117,6 +117,19 @@ function SubSectorPage({
     };
   }, []);
 
+  useEffect(() => {
+    const fetchData = async () => {
+      const res = await fetch(`/api/v0/subsector/${subsector}`);
+      if (!res.ok) {
+        throw new Error("Failed to fetch data");
+      }
+      const data = await res.json();
+      setRefNumber(data.data.referenceNumber);
+    };
+
+    fetchData();
+  }, []);
+
```

Comment:
Please use the a Redux Toolkit API hook (defined in `src/services/api/ts`) to fetch API data.

Reply:
Also just a note you can get the required subsector data from this line of code, should be on line 82. This way we don't have multiple api calls
```
const subSectorData = sectorData?.subSectors.find(
    (subsectorItem) => subsectorItem.subsectorId === subsector,
  );
```